### PR TITLE
Pull Request for Issue2091: Duplicate regions of interest

### PR DESCRIPTION
**Pull request by helfrij**
Added a new method containsRegionOfInterest() to BioXASBeamline. Successfully used this method when a user configuration is loaded to prevent a ton of duplicate ROIs from being added to the detector and scan configuration.

### CODE REVIEW

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you take a look at this pull request?

**Comment by dretrex** [comment]
Hey @helfrij, I know that what you've done technically fixes the problem, but I'd be interested to sit down with you sometime to figure out why it's adding duplicates in the first place because this could be happening on other beamlines.

File: source/beamline/AMXRFDetector.cpp [hunk 1]
```diff
@@ -322,9 +322,10 @@ void AMXRFDetector::addRegionOfInterest(const AMEmissionLine &emissionLine)
 
 	addRegionOfInterest(region);
 }
-
+#include <QDebug>
```
Comment (XuetaoLiu-CLS) [inline source/beamline/AMXRFDetector.cpp]: Do we want to keep this?

Comment (helfrij) [inline source/beamline/AMXRFDetector.cpp]: No, I don't think so. I'll take it out.

**Comment by XuetaoLiu-CLS** [comment]
Looks good to me. As @dretrex mentioned, this might not fix the root cause of the duplicated ROI. Hopefully we can find it.

Time Estimates: 5 minutes.

**Comment by dretrex** [comment]
Looks good.  We will need to make an issue to look into the possibility that this is a more general problem.
Time Estimate: 2 minutes

**Pull request merged**